Runtime/World/World.h

```diff
@@ -91,4 +114,28 @@ namespace Spartan
 		Scene_State m_state;
 		std::string m_name;
 	};
+
+    template<typename T>
+    inline std::shared_ptr<ComponentManager<T>> World::GetComponentManager()
+    {
+        VALIDATE_COMPONENT_TYPE(T);
+        ComponentType type = IComponent::TypeToEnum<T>();
+
+        std::shared_ptr<BaseComponentManager> manager = m_components_managers[type];
+
+        return std::static_pointer_cast<ComponentManager<T>>(manager);
+    }
+
+    template<typename Func>
+    inline void World::IterateManagers(Func f)
+    {
+        // Hardcoded types make it harder to write new components
```

Comment:
Hardcoded types mean that when a coder writes a new component, he'll have to visit all of this places and updated them. Do you think we can avoid them somehow ?